src/sentry/models/projectcodeowners.py

```diff
@@ -53,3 +63,100 @@ def get_codeowners_cached(self, project_id):
                 codeowners = False
             cache.set(cache_key, codeowners, READ_CACHE_DURATION)
         return codeowners or None
+
+    def update_schema(
+        self,
+    ):
+        """
+        Updating the schema goes through the following steps:
+
+        1. parsing the original codeowner file to get the external
+           teams, usernames, and emails
+        2. find the external actors for any usernames or team names
+        3. convert the codeowner file to the ownership syntax
+        4. convert the ownership syntax to the schema syntax
+
+        """
+
+        team_names, usernames, emails = parse_code_owners(self.raw)
+        external_actors = ExternalActor.objects.filter(
+            external_name__in=usernames + team_names,
+            organization=self.project.organization,
+        )
+
+        # Convert CODEOWNERS into IssueOwner syntax
+        issue_owner_rules, _ = codeowners_to_ownership_syntax(
+            self.raw, self.repository_project_path_config, self.project, external_actors, emails
+        )
+
+        # Convert IssueOwner syntax into schema syntax
+        schema = self.issue_syntax_to_schema(issue_owner_rules)
+        if schema:
+            self.schema = schema
+            self.save()
+
+    def issue_syntax_to_schema(self, raw):
+        """
+        Converts ownership syntax to the schema.
+        Duplicate code from ProjectOwnershipSerializer
```

Comment:
I think maybe converting the issue syntax to schema could be pulled out and put into another function that would live in the `grammar` file, and both here and the serializer could use it, but I didn't do that yet